client/src/components/app.jsx

```diff
@@ -15,13 +15,32 @@ class App extends React.Component {
       test: []
     }
     this.handleClick = this.handleClick.bind(this)
+    this.setToken = this.setToken.bind(this)
+    this.setUserId = this.setUserId.bind(this)
   }
 
   handleClick(e) {
     console.log(e.target.alt)
   }
 
+  setToken(value) {
+    this.setState({ token: value})
+  }
+
+  setUserId(value) {
+    this.setState({ user: value })
+  }
+
+
   render() {
+    if (!this.state.token) {
+      return (
+        <div>
+          <TopBar />
+          <Login setToken={this.setToken} setUserId={this.setUserId}/>
```

Comment:
Does this still render the rest of the app if you remove <Main />? Originally that was routing the rest of the application so want to confirm the rest will still work.

Reply:
hey @griseldasanchez for some reason I can't see what you put in the comment about what's missing.. but I think you're talking about the Main component?
![image](https://user-images.githubusercontent.com/60486755/133348670-ccdbe9d8-502e-477a-93d8-bcc41cd7d375.png)

Everything still works! I added this conditional above the return statement we had before. So when the user successfully logs in, a token is sent to app.jsx and the render goes down to the next statement where we have Main. I think that section of the pull request needs to be expanded to see the rest